### PR TITLE
Sync the marker line's position with the remote engine.

### PR DESCRIPTION
**Pull request by sebastianpoeplau**

### CODE REVIEW

File: src/Engine/Chats/ChatModel.cs [hunk 1]
```diff
@@ -47,6 +47,7 @@ public abstract class ChatModel : PermanentRemoteObject, ITraceable
         public  int                  Position { get; set; }
         public  IMessageBuffer       MessageBuffer { get; private set; }
         public  int                  MessagesSyncCount { get; set; }
+        public  DateTime             LastSeen { get; set; }
```
Comment (meebey) [inline src/Engine/Chats/ChatModel.cs]: Name the property LastSeenMessage (message is abstract in this context)

File: src/Frontend-GNOME/MainWindow.cs [hunk 2]
```diff
@@ -480,7 +480,7 @@ protected virtual void OnFocusOutEvent(object sender, EventArgs e)
                     return;
                 }
 
-                chatView.OutputMessageTextView.UpdateMarkerline();
+                chatView.UpdateLastSeen();
```
Comment (meebey) [inline src/Frontend-GNOME/MainWindow.cs]: This call needs to be in a background thread. See https://github.com/meebey/smuxi/blob/master/src/Frontend-GNOME/Views/Chats/ChatView.cs#L585

File: src/Frontend-GNOME/Views/Chats/ChatView.cs [hunk 3]
```diff
@@ -54,6 +54,7 @@ public abstract class ChatView : Gtk.EventBox, IChatView, IDisposable, ITraceabl
         private   MessageTextView    _OutputMessageTextView;
         private   ThemeSettings      _ThemeSettings;
         private   TaskQueue          _LastSeenHighlightQueue;
+        public    DateTime           SyncedLastSeen { get; private set; }
```
Comment (meebey) [inline src/Frontend-GNOME/Views/Chats/ChatView.cs]: Name it SyncedLastSeenMessage

File: src/Frontend-GNOME/MainWindow.cs (hunk 2 above)
Comment (meebey) [inline src/Frontend-GNOME/MainWindow.cs]: Hm this should happen inside UpdateLastSeenMessage

File: src/Frontend-GNOME/Views/Chats/ChatView.cs [hunk 4]
```diff
@@ -578,6 +582,14 @@ public virtual void Populate()
             _IsSynced = true;
         }
         
+        public virtual void UpdateLastSeen() {
```
Comment (meebey) [inline src/Frontend-GNOME/Views/Chats/ChatView.cs]: CS: { needs newline

File: src/Frontend-GNOME/Views/Chats/ChatView.cs [hunk 5]
```diff
@@ -578,6 +582,14 @@ public virtual void Populate()
...
+            _OutputMessageTextView.UpdateMarkerline();
+            if (_OutputMessageTextView.LastMessage != null) {
```
Comment (meebey) [inline src/Frontend-GNOME/Views/Chats/ChatView.cs]: This is racy, use a local var (once you add the background thread, it will be clear why)

**Comment by meebey** [comment]
I am done with the review

File: src/Frontend-GNOME/Views/Chats/ChatView.cs [hunk 6]
```diff
@@ -519,6 +520,8 @@ public virtual void Sync()
 
             // REMOTING CALL
             SyncedLastSeenHighlight = _ChatModel.LastSeenHighlight;
+            // REMOTING CALL
+            SyncedLastSeenMessage = _ChatModel.LastSeenMessage;
```
Comment (meebey) [inline src/Frontend-GNOME/Views/Chats/ChatView.cs]: This needs to be in a if() block that check for Smuxi engine version

File: src/Frontend-GNOME/Views/Chats/ChatView.cs [hunk 7]
```diff
@@ -578,6 +582,23 @@ public virtual void Populate()
             _IsSynced = true;
         }
         
+        public virtual void UpdateLastSeenMessage()
+        {
+            _OutputMessageTextView.UpdateMarkerline();
+            
+            var lastSeenMessage = _OutputMessageTextView.LastMessage;
+            if (lastSeenMessage != null) {
+                ThreadPool.QueueUserWorkItem(delegate {
```
Comment (meebey) [inline src/Frontend-GNOME/Views/Chats/ChatView.cs]: Smuxi engine version check

**Comment by meebey** [comment]
Your commit messages need to use the module prefix
Frontend-GNOME: $commit_message

File: src/Frontend-GNOME/Views/Chats/ChatView.cs (hunk 6 above)
Comment (sebastianpoeplau) [inline src/Frontend-GNOME/Views/Chats/ChatView.cs]: Will do. You mentioned that merely checking for >= 0.12 will break dev builds. Should I bump the version to 0.12.1 then? (And if so, where can I do it? :) )

**Comment by sebastianpoeplau** [comment]
Sure. I'll squash the commits as soon as you deem the code ready to merge and reformat the commit message.

File: src/Frontend-GNOME/Views/Chats/ChatView.cs (hunk 6 above)
Comment (meebey) [inline src/Frontend-GNOME/Views/Chats/ChatView.cs]: Leave the version and check for >= new Version(0, 12) in the code

**Comment by sebastianpoeplau** [comment]
Is anything else missing for this to be merged?

File: src/Frontend-GNOME/Views/Chats/ChatView.cs [hunk 8]
```diff
@@ -578,6 +585,25 @@ public virtual void Populate()
             _IsSynced = true;
         }
         
+        public virtual void UpdateLastSeenMessage()
+        {
+            _OutputMessageTextView.UpdateMarkerline();
+            
+            if (Frontend.EngineVersion >= new Version(0, 12)) {
```
Comment (meebey) [inline src/Frontend-GNOME/Views/Chats/ChatView.cs]: CS: can you use return here instead, this method has no other purpose from here than to push the change to the engine

File: src/Frontend-GNOME/Views/Chats/ChatView.cs [hunk 9]
```diff
@@ -578,6 +585,25 @@ public virtual void Populate()
...
+                var lastSeenMessage = _OutputMessageTextView.LastMessage;
+                if (lastSeenMessage != null) {
```
Comment (meebey) [inline src/Frontend-GNOME/Views/Chats/ChatView.cs]: CS: same here, please use return

**Comment by meebey** [comment]
code functionality is good, I like how this small code change brings such a powerful feature

**Pull request merged**